orca-clouddriver/src/main/groovy/com/netflix/spinnaker/orca/clouddriver/pipeline/servergroup/strategies/RollingRedBlackStrategy.groovy

```diff
@@ -120,10 +118,29 @@ class RollingRedBlackStrategy implements Strategy, ApplicationContextAware {
       SyntheticStageOwner.STAGE_AFTER
     )
 
+    // Get source ASG from prior determineSourceServerGroupTask
     def source = null
+
     try {
-      source = getSource(targetServerGroupResolver, stageData, baseContext)
-    } catch (TargetServerGroup.NotFoundException e) {}
+      StageData.Source sourceAsg
+
+      Stage parentCreateAsgStage = stage.ancestors()
+                                        .find() { it.type == CreateServerGroupStage.PIPELINE_CONFIG_TYPE }
```

Comment:
is this what `spotlessApply` gave you ... seems like weird formatting but 🤷‍♀ 

more `s/Asg/ServerGroup`

Reply:
I'm always tentative with `ancestors()` looking only for stage type name ... are we confident this is only going to find what we want?

(vs. say looking at the parent of this stage and then it's synthetic children or something)

Reply:
as in it could find a CreateServerGroup stage that's completely irrelevant.. that's fair... (it shouldn't but it's brittle, you're right). let me think about this a bit how to clean it up

Reply:
I made a helper that only traverses "real" parents, what do you think?